### PR TITLE
Update demo, examples and configs to react-map-gl 3 and webpack 2

### PR DESCRIPTION
**Pull request by ibgreen**

### CODE REVIEW

File: demo/README.md [hunk 1]
```diff
@@ -3,7 +3,7 @@ This is the source code of the demo website for deck.gl.
 ## Developing
 
     npm install
-    export MAPBOX_ACCESS_TOKEN=<insert_your_token> && npm start
+    export MapboxAccessToken=<insert_your_token> && npm start
```
Comment (Apercu) [inline demo/README.md]: why changing the env name?

File: demo/.babelrc [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  presets: [
```
Comment (Apercu) [inline demo/.babelrc]: neat: can be a one liner in the separate `babelrc`

File: demo/README.md (hunk 1 above)
Comment (Pessimistress) [inline demo/README.md]: We used to support both `MAPBOX_ACCESS_TOKEN` and `MapboxAccessToken` in react-map-gl, v3 dropped the former

File: src/controllers/map-controls.js [hunk 3]
```diff
@@ -59,9 +59,9 @@ const propTypes = {
 
   constraints: PropTypes.object,
 
-  perspectiveEnabled: PropTypes.bool,  // Enables perspective control event handling
+  dragToRotate: PropTypes.bool,  // Enables perspective control event handling
```
Comment (Pessimistress) [inline src/controllers/map-controls.js]: Why do we have a copy of react-map-gl components in deck.gl?

File: demo/.babelrc (hunk 2 above)
Comment (ibgreen) [inline demo/.babelrc]: > neat: can be a one liner in the separate babelrc

Actually the separate babelrc still doesn't work, will remove.

File: src/controllers/map-controls.js (hunk 3 above)
Comment (ibgreen) [inline src/controllers/map-controls.js]: > Why do we have a copy of react-map-gl components in deck.gl?

I used to do integration experiments with non-mapbox maps, and I was also investigating building on a family of controllers that would follow similar principles.

It is another chapter in the never ending story of what should reasonably be common components between these two frameworks, but with no good place to put them.

We can remove it for now. Another PR though.

**Pull request merged**